src/ape_http/providers.py

```diff
@@ -174,20 +174,19 @@ def connect(self):
 
             if not self._web3.isConnected():
                 self._geth.disconnect()
-                raise ConnectionError("Unable to connect web3 to geth.")
+                raise ConnectionError("Unable to connect to locally running geth.")
 
         self._web3.eth.set_gas_price_strategy(rpc_gas_price_strategy)
 
+        if self.network.name not in ("mainnet", "ropsten"):
```

Comment:
I would flip this, because I don't believe local networks need this either, and neither would some other testnets for PoW chains
```suggestion
        if self.network.name in ("rinkeby", "goerli", "kovan"):
```

Also might want to do this via chain_id.... :thinking:

Reply:
Done and Done..

Wait, would local PoA networks need this though? I just added some code that checks the consensus via `admin.node_info()` and will set this middleware if `clique` was configured